csharp/src/Apache.Arrow/Extensions/SpanExtensions.cs

```diff
@@ -14,18 +14,65 @@
 // limitations under the License.
 
 using System;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
 namespace Apache.Arrow
 {
     public static class SpanExtensions
     {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Span<T> CastTo<T>(this Span<byte> span)
             where T: struct =>
             MemoryMarshal.Cast<byte, T>(span);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ReadOnlySpan<T> CastTo<T>(this ReadOnlySpan<byte> span)
             where T: struct =>
                 MemoryMarshal.Cast<byte, T>(span);
+
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void CopyToFix<T>(this ReadOnlySpan<T> source, Span<T> target)
```

Comment:
I don't think these methods should be `public`. We are going to want to remove them in the future when the underlying bug is fixed.

Reply:
made them internal